.sql/data.sql

```diff
@@ -0,0 +1,246 @@
+-- phpMyAdmin SQL Dump
+-- version 4.6.6deb1+deb.cihar.com~xenial.2
+-- https://www.phpmyadmin.net/
+--
+-- Host: localhost
+-- Czas generowania: 24 Paź 2018, 00:34
+-- Wersja serwera: 5.7.20-0ubuntu0.16.04.1
+-- Wersja PHP: 7.2.11-2+ubuntu16.04.1+deb.sury.org+1
+
+SET SQL_MODE = "NO_AUTO_VALUE_ON_ZERO";
+SET time_zone = "+00:00";
+
+--
+-- Baza danych: `adfind`
+--
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `Category`
+--
+
+CREATE TABLE `Category` (
+  `id` int(11) NOT NULL,
+  `name` varchar(255) NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+--
+-- Zrzut danych tabeli `Category`
+--
+
+INSERT INTO `Category` (`id`, `name`) VALUES
+(1, 'Photos'),
+(2, 'Pictograms'),
+(3, 'Illustrations'),
+(4, 'Documents');
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `File`
+--
+
+CREATE TABLE `File` (
+  `id` int(11) NOT NULL,
+  `name_add` varchar(255) NOT NULL,
+  `name_view` varchar(255) DEFAULT NULL,
+  `name_hash` varchar(255) NOT NULL,
+  `subcategory_id` int(11) DEFAULT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `Format`
+--
+
+CREATE TABLE `Format` (
+  `id` int(11) NOT NULL,
+  `name` varchar(255) NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+--
+-- Zrzut danych tabeli `Format`
+--
+
+INSERT INTO `Format` (`id`, `name`) VALUES
+(1, 'jpeg'),
+(2, 'png'),
+(3, 'svg'),
+(4, 'eps'),
+(5, 'txt'),
+(6, 'doc'),
+(7, 'ppt'),
+(8, 'xls');
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `MapFormat`
+--
+
+CREATE TABLE `MapFormat` (
+  `id` int(11) NOT NULL,
+  `file_id` int(11) DEFAULT NULL,
+  `format_id` int(11) DEFAULT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `MapTag`
+--
+
+CREATE TABLE `MapTag` (
+  `id` int(11) NOT NULL,
+  `file_id` int(11) DEFAULT NULL,
+  `tag_id` int(11) DEFAULT NULL
```

Comment:
Dorzuć tutaj pojedynczy! klucz unikalny na pola tag_id, file_id. Kolejność jest istotna: https://dev.mysql.com/doc/refman/5.7/en/multiple-column-indexes.html . Klucze powinny być tak zbudowane, aby kolumny zaczynając od lewej jak najwięcej "zabierały" do przeszukiwania następnej w kolejności. Przykładowo, zakładam że mniej jest plików niż tagów. W pierwszej więc kolejności zacząłbym filtrować po tagach, aby w drugim kroku mieć mniej plików do przeszukania. Jeśli miałabyś np 10k plików oraz 5 tagów z czego każdy miałby po 2k plików to sytuacja wyglądałaby tak, że najpierw iterujesz po 5 tagach -> wybierasz jeden -> iterujesz po 2k plików. W odwrotnej sytuacji iterowałabyś od razu po 10k rekordów, a na samym końcu dopiero po tagach (czyli w tym przypadku po jednym)